seclea_ai/internal/file_processor.py

```diff
@@ -146,32 +280,230 @@ def _send_dataset(
                     project=project,
                     organization=self._organization,
                 )
-
         else:
             raise ValueError(
                 f"Response Status code {response.status}, expected: 201. \n f'There was some issue uploading the dataset: {response.text()}' - {response.reason} - {response.text}"
             )
 
+    def upload_training_run(
+        self,
+        model,
+        project,
+        train_dataset: DataFrame,
+        test_dataset: DataFrame = None,
+        val_dataset: DataFrame = None,
+    ) -> None:
+        """
+        Takes a model and extracts the necessary data for uploading the training run.
+
+        :param model: An ML Model instance. This should be one of {sklearn.Estimator, xgboost.Booster, lgbm.Boster}.
+
+        :param train_dataset: DataFrame The Dataset that the model is trained on.
+
+        :param test_dataset: DataFrame The Dataset that the model is trained on.
+
+        :param val_dataset: DataFrame The Dataset that the model is trained on.
+
+        :return: None
+
+        Example::
+
+            >>> seclea = SecleaAI(project_name="Test Project")
+            >>> dataset = pd.read_csv(<dataset_name>)
+            >>> model = LogisticRegressionClassifier()
+            >>> model.fit(X, y)
+            >>> seclea.upload_training_run(
+                    model,
+                    framework=seclea_ai.Frameworks.SKLEARN,
+                    dataset_name="Test Dataset",
+                )
+        """
+        self._auth_service.authenticate(self._transmission)
+
+        # validate the splits? maybe later when we have proper Dataset class to manage these things.
+        dataset_pks = [
+            str(hash(pd.util.hash_pandas_object(dataset).sum() + project))
+            for dataset in [train_dataset, test_dataset, val_dataset]
+            if dataset is not None
+        ]
+
+        model_name = model.__class__.__name__
+
+        framework = self._get_framework(model)
+
+        # check the model exists upload if not
+        model_type_pk = self._set_model(model_name=model_name, framework=framework)
+
+        # check the latest training run
+        training_runs_res = self._transmission.get(
+            "/collection/training-runs",
+            query_params={
+                "project": self._project,
+                "model": model_type_pk,
+                "organization": self._organization,
+            },
+        )
+        training_runs = training_runs_res.json()
+
+        # Create the training run name
+        largest = -1
+        for training_run in training_runs:
+            num = int(training_run["name"].split(" ")[2])
+            if num > largest:
+                largest = num
+        training_run_name = f"Training Run {largest + 1}"
+
+        # extract params from the model
+        params = framework.value.get_params(model)
+
```

Comment:
This should still be in SecleaAI for now I think.

Reply:
Ok

Reply:
I have refactored, resolving this.